### PR TITLE
docs: Document node field selector. Closes #2860

### LINKED ISSUE #2860: Need documentation on "nodeFieldSelector"
We need to add some documentation regarding `nodeFieldSelector`s in CLI commands. Ideally, it would be a doc under `docs/` something like `node-field-selector.md` following the same template of other docs. The docs should contain a list of all available "node selector fields" in it.

I think the closest example for this would be something like https://github.com/argoproj/argo/blob/master/docs/default-workflow-specs.md

### PR DESCRIPTION
**Pull request by mark9white**
Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [X] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [X] I've signed the CLA.
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [X] My builds are green. Try syncing with master if they are not. 
* [X] My organization is added to USERS.md.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -29,6 +30,7 @@ Some use-case specific documentation is available:
 * [Links](links.md)
 * [Managed Namespace](managed-namespace.md)
 * [Prometheus Metrics](metrics.md)
+* [node-field-selector](node-field-selector.md)
```
Comment (simster7) [inline docs/README.md:33]:
```suggestion
* [Node Field Selectors](node-field-selector.md)
```

File: docs/README.md [hunk 2]
```diff
@@ -18,6 +18,7 @@ Some use-case specific documentation is available:
 * [Contributing](CONTRIBUTING.md)
 * [Argo Workflow Architecture](architecture.md)
 * [Argo Server](argo-server.md)
+* [Asynchronous job pattern](async-pattern.md)
```
Comment (simster7) [inline docs/README.md:21]:
```suggestion
* [Asynchronous Job Pattern](async-pattern.md)
```

File: docs/async-pattern.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+# Asynchronous job pattern
```
Comment (simster7) [inline docs/async-pattern.md:1]:
```suggestion
# Asynchronous Job Pattern
```

File: docs/node-field-selector.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# node-field-selector
```
Comment (simster7) [inline docs/node-field-selector.md:1]:
```suggestion
# Node Field Selectors
```

File: docs/node-field-selector.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+The field can be any of:
+
+| Field | Description|
+|----------|------------|
+| displayName | Display name of the node |
+| templateName | Template name of the node |
+| phase | Phase status of the node - eg Running |
+| templateRef.name | The name of the WorkflowTemplate the node is referring to |
+| templateRef.template | The template within the WorkflowTemplate the node is referring to |
+| inputs.parameters.xxxx.value | The value of input parameter xxxx |
```
Comment (simster7) [inline docs/node-field-selector.md:30]:
```suggestion
| inputs.parameters.<NAME>.value | The value of input parameter NAME |
```

File: docs/async-pattern.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+## Introduction
+
+If triggering an external job (eg an Amazon EMR job) from Argo that does not run to completion in a container, there are two options:
```
Comment (simster7) [inline docs/async-pattern.md:5]: This doc is great

File: pkg/apiclient/workflow/workflow.proto [hunk 7]
```diff
@@ -43,7 +43,7 @@ message WorkflowResubmitRequest {
 message WorkflowRetryRequest {
     string name = 1;
     string namespace = 2;
-    bool restartSuccesful = 3;
```
Comment (alexec) [inline pkg/apiclient/workflow/workflow.proto:46]: Is this a breaking change? Has this gone GA yet?

Comment (mark9white) [reply pkg/apiclient/workflow/workflow.proto:46]: It is breaking, but it's part of v2.8 and hasn't gone GA yet.

Comment (alexec) [reply pkg/apiclient/workflow/workflow.proto:46]: 👍

File: docs/node-field-selector.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
+# Node Field Selectors
+
+![GA](assets/ga.svg)
```
Comment (alexec) [inline docs/node-field-selector.md:3]: I think this should be `alpha` or `beta`.

Comment (simster7) [reply docs/node-field-selector.md:3]: I'd say `alpha`

**Comment by mark9white** [comment]
@simster7 done

**Comment by alexec** [comment]
Please run `make lint` to fix CI.

**Comment by mark9white** [comment]
@alexec lint issue should be fixed

**Pull request merged**